www/src/js/views/settings-time.js

```diff
@@ -5,7 +5,147 @@ var template = require('../templates/settings-time.hbs');
 module.exports = Backbone.Marionette.ItemView.extend({
   className: 'b-settings__time',
 
-  template: function() {
-    return template();
+  events: {
+    'change #date-picker': 'dateChanged',
+    'change #time-picker': 'timeChanged',
+    'change #ntp-active-checkbox': 'ntpActiveCheckboxChanged',
+    'change #ntp-server-name': 'ntpServerNameChanged',
+    'change #time-zone-select': 'timezoneSelectChanged'
+  },
+
+  initialize: function() {
+  },
+
+  template: function(model) {
+    return template(model);
+  },
+
+  dateChanged: function() {
+    var datePicker = this.$('#date-picker');
+    this.model.save({'date': datePicker.val()}, {patch: true});
+  },
+
+  timeChanged: function() {
+    var timePicker = this.$('#time-picker');
+    this.model.save({'time': timePicker.val()}, {patch: true});
+  },
+
+  ntpActiveCheckboxChanged: function() {
+    var checked = this.$('#ntp-active-checkbox').prop('checked');
+
+    this.$('#ntp-server-name').prop('disabled', !checked)
+    this.$('#date-picker').prop('disabled', checked);
+    this.$('#time-picker').prop('disabled', checked);
+  },
+
+  ntpServerNameChanged: function() {
+    var ntpServerName = this.$('#ntp-server-name').val();
+
+    if (ntpServerName !== this.model.get('ntpServer')) {
+      this.model.save({'ntpServer': ntpServerName}, {patch: true});
+    }
+  },
+
+  timezoneSelectChanged: function() {
+    var selected = this.$('#time-zone-select').val();
+    if (selected != this.model.get('timezone')) {
+      this.model.save({'timezone': selected}, {patch: true});
+    }
+  },
+
+  onRender: function() {
+    var haveServer = this.model.get('ntpServer') !== "";
+
+    this.$('#ntp-active-checkbox').prop('checked', haveServer);
+    this.$('#ntp-server-name').prop('disabled', !haveServer)
+    this.$('#date-picker').prop('disabled', haveServer);
+    this.$('#time-picker').prop('disabled', haveServer);
+
+    var dropDown = this.$('#time-zone-select');
+    dropDown.append([
+          '<option value="-12">(UTC-12:00) International Date Line West</option>',
```

Comment:
Just curious, but how is i18n usually done in JS applications like these? I guess its fine for now to only support english but I'm curious what the longer term strategy here is :)